library/network_connections.py

```diff
@@ -803,8 +803,12 @@ def connection_create(self, connections, idx, connection_current=None):
             s_con.set_property(
                 NM.SETTING_CONNECTION_TYPE, NM.SETTING_WIRED_SETTING_NAME
             )
-            s_wired = self.connection_ensure_setting(con, NM.SettingWired)
-            s_wired.set_property(NM.SETTING_WIRED_MAC_ADDRESS, connection["mac"])
+            if not connection["mac"]:
+                s_wired = self.connection_ensure_setting(con, NM.SettingWired)
+                s_wired.set_property(NM.SETTING_WIRED_MAC_ADDRESS, connection["mac"])
+            else:
+                match = NM.SettingMatch()
+                match.set_property(NM.SETTING_MATCH_PATH, connection["pci"])
         elif connection["type"] == "infiniband":
```

Comment:
test error:
```
 "changed": false, "msg": "fatal error: expected sequence", "stderr": "\n", "stderr_lines": [""]}
 ```
if `NM.SettingMatch()` == `NM.SettingMatch.new()`, I should be able to use `set_property()` function. (`NMSetting.set_property()`)

@thom311 , do you know where was wrong?

Reply:
I suspect you'll have to use `hasattr` here too unless @thom311 @cathay4t have a better idea

Reply:
`NMSettingMatch:path` was included since NM-1.26

Reply:
> Does this PR need to be documented in the README.md?

yes, we also need to update the `README.md` to show that how user can use this feature

Reply:
> I suspect you'll have to use `hasattr` here too unless @thom311 @cathay4t have a better idea

It might be better to check for NetworkManager support in the ArgValidator so an unsupported configuration will fail early.

Reply:
I added the check for NM support in ArgValidator, see  https://github.com/linux-system-roles/network/blob/aa27d56c6d67d7b883b5415a3a972678642ab835/module_utils/network_lsr/argument_validator.py#L1823